src/linker/step.py

```diff
@@ -16,14 +15,3 @@ class Step:
 
     name: str
     validate_file: Callable = validate_dummy_file
```

Comment:
This is a confusing name for a Callable now that we actually have validate files. Maybe `validation_function` or `validator`?